pkg/operator/controller/ingress/status.go

```diff
@@ -556,6 +574,50 @@ func computeIngressUpgradeableCondition(ic *operatorv1.IngressController, deploy
 	}
 }
 
+// checkDefaultCertificate returns an error value indicating whether the default
+// certificate is safe for upgrades.  In particular, if the current default
+// certificate specifies a Subject Alternative Name (SAN) for the ingress
+// domain, then it is safe to upgrade, and the return value is nil.  Otherwise,
+// if the certificate has a legacy Common Name (CN) for the ingress domain and
+// no SAN for the same, then the return value is an error indicating that the
+// certificate must be replaced by one with a SAN before upgrading is allowed.
+// This check is necessary because OpenShift 4.10 is built using Go 1.17, which
+// rejects certificates without SANs.  Note that this function only checks the
+// validity of the certificate insofar as it affects upgrades.  See
+// <https://bugzilla.redhat.com/show_bug.cgi?id=2057762>.  This check can be
+// removed after OpenShift 4.9.
```

Comment:
This is in 4.9, right?

```suggestion
// <https://bugzilla.redhat.com/show_bug.cgi?id=2057762>.  
```

Reply:
Yes, this is the 4.9 backport.  Usually I keep the backport as close as possible to the original change (in this case, #708) to minimize risk of breaking something and to simplify future backports.

Reply:
I was thinking that if someone saw "This check can be removed after OpenShift 4.9." that they might remove it in a later 4.9 version.  Or it would at least cause some questions.

Reply:
Hm, I guess someone could interpret "after OpenShift 4.9" as including newer z-stream releases _of_ 4.9.  Maybe I should have said "in OpenShift 4.10 or later" in the original commit.  However, even if someone interpreted "after OpenShift 4.9" in that way, I think the risk that someone would remove this in a 4.9 z-stream is low since we have extra levels of review and scrutiny for backports.

Reply:
I've amended the comment.